lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1948,6 +1973,62 @@ bool QuakeBridgeVisitor::VisitCXXConstructExpr(clang::CXXConstructExpr *x) {
         assert((isa<quake::RefType, quake::VeqType>(tosTy)));
         return true;
       }
+
+      // We check for vector constructors with 2 args, the first
+      // could be an initializer_list or an integer, while the
+      // second is the allocator
+      if (ctorName == "vector" && x->getNumArgs() == 2) {
+        // This is a std::vector constructor, first we'll check if it
+        // is constructed from a constant initializer list, in that case
+        // we'll have a AllocaOp at the top of the stack that allocates a
+        // ptr<array<TxC>>, where C is constant / known
+        auto desugared = x->getArg(0)->getType().getCanonicalType();
+        if (auto recordType =
+                dyn_cast<clang::RecordType>(desugared.getTypePtr())) {
+          if (recordType->getDecl()->getName().equals("initializer_list")) {
+            auto allocation = popValue();
+            if (auto ptrTy = dyn_cast<cc::PointerType>(allocation.getType())) {
+              if (auto arrayTy =
+                      dyn_cast<cc::ArrayType>(ptrTy.getElementType())) {
+                if (auto definingOp = allocation.getDefiningOp<cc::AllocaOp>())
+                  return pushValue(builder.create<cc::StdvecInitOp>(
+                      loc, cc::StdvecType::get(arrayTy.getElementType()),
+                      allocation, definingOp.getSeqSize()));
+              }
+            }
+          }
+        }
+
+        // Next check if its created from a size integer
+        // Let's do a check on the first argument, make sure that when
+        // we peel off all the typedefs that it is an integer
+        if (auto builtInType =
+                dyn_cast<clang::BuiltinType>(desugared.getTypePtr())) {
+          if (builtInType->isInteger() &&
+              peekValue().getType().isIntOrFloat()) {
```

Comment:
I was thrown by this. Instead of `isIntOrFloat`, maybe we should do `isa<IntegerType>(peek.getType())`?

Reply:
Fixed.